mypy/fastparse2.py

```diff
@@ -489,18 +489,18 @@ def visit_With(self, n: ast27.With) -> WithStmt:
 
     @with_line
     def visit_Raise(self, n: ast27.Raise) -> RaiseStmt:
-        e = None
-        if n.type is not None:
-            e = n.type
-
-            if n.inst is not None and not (isinstance(n.inst, ast27.Name) and n.inst.id == "None"):
-                if isinstance(n.inst, ast27.Tuple):
-                    args = n.inst.elts
+        if n.type is None:
+            e = None
+        else:
+            if n.inst is None:
+                e = self.visit(n.type)
+            else:
+                if n.tback is None:
+                    e = TupleExpr([self.visit(n.type), self.visit(n.inst)])
```

Comment:
BTW did you know that type_check_raise() in checker.py doesn't actually check the types when the arg is a tuple? Another issue is that `raise (x, y)` does not mean the same as `raise x, y`! I guess that's for a later PR.